### PR TITLE
Update run.sh

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [x]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [x]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]

File: docker/test/stateless/run.sh [hunk 1]
```diff
@@ -249,7 +249,7 @@ function run_tests()
     try_run_with_retry 10 clickhouse-client -q "insert into system.zookeeper (name, path, value) values ('auxiliary_zookeeper2', '/test/chroot/', '')"
 
     set +e
-    timeout -s TERM --preserve-status 120m clickhouse-test --testname --shard --zookeeper --check-zookeeper-session --hung-check --print-time \
+    timeout -s KILL --preserve-status 120m clickhouse-test --testname --shard --zookeeper --check-zookeeper-session --hung-check --print-time \
```
Comment (alexey-milovidov) [inline docker/test/stateless/run.sh:252]: But why?

Comment (kssenii) [reply docker/test/stateless/run.sh:252]: Previous way did not help with "Check timeout expired" resulting in zero server logs and no way to debug what it happened

**Comment by jkartseva** [comment]
There is also `--kill-after` option:
 ```
  -k, --kill-after=DURATION
                 also send a KILL signal if COMMAND is still running
                   this long after the initial signal was sent
```

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 5a12659f43f74aa501610404c4b2ee6b1b4a02c9 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/66290/5a12659f43f74aa501610404c4b2ee6b1b4a02c9/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>ClickBench</td><td>Runs [ClickBench](https://github.com/ClickHouse/ClickBench/) with instant-attach table</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/44dc820751996ccd8cf2df3091b1b970bf86bc27/clickbench__aarch64_.html">✅ success</a></td></tr>
<tr><td>Compatibility check</td><td>Checks that <b>clickhouse</b> binary runs on distributions with old libc versions. If it fails, ask a maintainer for help</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/44dc820751996ccd8cf2df3091b1b970bf86bc27/compatibility_check__aarch64_.html">✅ success</a></td></tr>
<tr><td>Docker keeper image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/44dc820751996ccd8cf2df3091b1b970bf86bc27/docker_keeper_image.html">✅ success</a></td></tr>
<tr><td>Docker server image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/44dc820751996ccd8cf2df3091b1b970bf86bc27/docker_server_image.html">✅ success</a></td></tr>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/4add9db84859bff7410cf934a3904b0414e36e51/docs_check.html">✅ success</a></td></tr>
<tr><td>Fast test</td><td>Normally this is the first check that is ran for a PR. It builds ClickHouse and runs most of <a href="https://clickhouse.com/docs/en/development/tests#functional-tests">stateless functional tests</a>, omitting some. If it fails, further checks are not started until it is fixed. Look at the report to see which tests fail, then reproduce the failure locally as described <a href="https://clickhouse.com/docs/en/development/tests#functional-test-locally">here</a></td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/44dc820751996ccd8cf2df3091b1b970bf86bc27/fast_test.html">✅ success</a></td></tr>
<tr><td>Install packages</td><td>Checks that the built packages are installable in a clear environment</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/44dc820751996ccd8cf2df3091b1b970bf86bc27/install_packages__aarch64_.html">✅ success</a></td></tr>
<tr><td>Integration tests</td><td>The integration tests report. In parenthesis the package type is given, and in square brackets are the optional part/total tests</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/44dc820751996ccd8cf2df3091b1b970bf86bc27/integration_tests__aarch64__[1_6].html">✅ success</a></td></tr>
<tr><td>Performance Comparison</td><td>Measure changes in query performance. The performance test report is described in detail <a href="https://github.com/ClickHouse/ClickHouse/tree/master/docker/test/performance-comparison#how-to-read-the-report">here</a>. In square brackets are the optional part/total tests</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/44dc820751996ccd8cf2df3091b1b970bf86bc27/performance_comparison__release__[1_4]/report.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/44dc820751996ccd8cf2df3091b1b970bf86bc27/unit_tests__asan_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by kssenii** [comment]
Potentially the issue will be fixed by https://github.com/ClickHouse/ClickHouse/pull/66036. Will reopen the PR if it is not fixed.

**Comment by kssenii** [comment]
> Will reopen the PR if it is not fixed.

It is not.

**Comment by kssenii** [comment]
> Stateless tests (debug) [1/2] — Tests are not finished
Stateless tests (debug, s3 storage) [1/2] — Tests are not finished

Failed assertion: https://github.com/ClickHouse/ClickHouse/issues/66390

> Stateless tests (debug) [2/2] — Tests are not finished
Stateless tests (debug, s3 storage) [2/2] — Tests are not finished

Also abort, should be fixed by https://github.com/ClickHouse/ClickHouse/pull/66312

**Pull request merged**